test/cppsuite/test_harness/workload/database_operation.cxx

```diff
@@ -161,27 +161,26 @@ database_operation::insert_operation(thread_context *tc)
         /* Collection cursor. */
         auto &cc = ccv[counter];
         while (tc->transaction.active() && tc->running()) {
-            /* Insert a key value pair. */
-            bool rollback_required = tc->insert(cc.cursor, cc.coll.id, start_key + added_count);
-            if (!rollback_required) {
+            /* Insert a key value pair, rolling back the transaction if required. */
+            if (!tc->insert(cc.cursor, cc.coll.id, start_key + added_count)) {
```

Comment:
Can we check for `true` instead here? Makes more sense to me but it's only my opinion...

Reply:
Wouldn't we check for false here? I.e. tc->insert == false?

Reply:
Sorry I was not clear. Can we do something like this:
```
if(tc->insert(cc.cursor, cc.coll.id, start_key + added_count)) {
<code_when_insert_is_successful>
} else {
added_count = 0;
tc->transaction.rollback();
}
```